src/main/java/com/benmorant/game/holycraap/model/entity/People.java

```diff
@@ -33,7 +28,125 @@
   private int currentMp;
   private int mpMax;
 
-  @OneToMany
-  @JoinColumn(name = "people_id")
-  private List<Item> inventory;
+  @OneToMany(mappedBy = "people")
+  @JsonManagedReference
+  private List<Item> inventory = new ArrayList<>();
+
+  public People() {}
+
+  public People(@NonNull String name) {
+    this.name = name;
+  }
+
+  /** Constructeur surchargé pour People (moins l'ID, auto-généré .) */
+  public People(@NonNull String name, int currentHp, int hpMax, int currentMp, int mpMax) {
+    this.name = name;
+    this.currentHp = currentHp;
+    this.hpMax = hpMax;
+    this.currentMp = currentMp;
+    this.mpMax = mpMax;
+  }
+
+  public void addItem(Item item) {
+    inventory.add(item);
+    item.setPeople(this);
```

Comment:
You should do one or the other not both. Remember what matters here is what is happening in the database.

Reply:
I just followed https://vladmihalcea.com/jpa-hibernate-synchronize-bidirectional-entity-associations/

"However, we still need to have both sides in sync as otherwise, we break the Domain Model relationship consistency, and the entity state transitions are not guaranteed to work unless both sides are properly synchronized.

For this reason, the Post entity defines the addComment and removeComment entity state synchronization methods."

(and I find it more logical and more practical, somehow )

Reply:
I understand why you do that, but database-wise, setting that field twice won't change the outcome, since it's only one column in the database. Also this code is incomplete and still breaks the consistency with the database as it does not cover adding an object that is already in the `inventory`.

Two solutions to that, either you make `inventory` an `ArrayList` and/or you modify the code of `addItem` to check that the `item` is not in `inventory` yet. `inventory` being declared as a `List<Item>`, the `addItem` method should not assume `inventory` checks for unicity.

Reply:
Now Inventory is an Arraylist (it already was before) and I modified this method :   public void addItem(Item item) {
    if (!inventory.contains(item)) {
      inventory.add(item);
    }
    item.setPeople(this);
  }